agents-api/agents_api/activities/summarization.py

```diff
@@ -160,28 +164,56 @@ async def run_prompt(
 @activity.defn
 async def summarization(session_id: str) -> None:
     session_id = UUID(session_id)
```

Comment:
[only in adaptive mode] add some logic to handle messages that contain images. If `message.image_url.detail != "low"` then produce a new entry that "summarizes" that entry by setting `detail` to `"low"`

Reply:
apply this first, see if the token threshold is still being breached and then apply the summarize step. we should do this before the trim operation too -- only apply the `trim` operation if the summarize and entities step result is still greater than the token threshold